File: Backend/sms-backend/src/schemas/academics/section.py
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict

from src.schemas.base.base import TimestampSchema, TenantSchema


class SectionBase(BaseModel):
    """Base schema for Section model."""
    name: str
    description: Optional[str] = None
    is_active: bool = True
    capacity: int
    grade_id: UUID
    class_teacher_id: Optional[UUID] = None


class SectionCreate(SectionBase):
    """Schema for creating a new section."""
    pass


class SectionUpdate(BaseModel):
    """Schema for updating a section."""
    name: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None
    capacity: Optional[int] = None
    grade_id: Optional[UUID] = None
    class_teacher_id: Optional[UUID] = None


class SectionInDB(SectionBase, TenantSchema):
    """Schema for Section model in database."""
    model_config = ConfigDict(from_attributes=True)


class Section(SectionInDB):
    """Schema for Section model response."""
    pass


class SectionWithDetails(Section):
    """Schema for Section with additional details."""
    grade_name: str
    class_teacher_name: Optional[str] = None

File: Frontend/sms-client/src/types/section.ts
export interface SectionBase {
  name: string;
  grade_id: string;
  capacity?: number;
  description?: string;
}


export type SectionCreate = SectionBase;

export type SectionUpdate = Partial<SectionBase>;

export interface Section extends SectionBase {
  id: string;
  grade_name: string;
  created_at: string;
  updated_at: string;
}

export interface SectionWithDetails extends Section {
  student_count?: number;
  class_count?: number;
}

export interface SectionResponse {
  sections: Section[];
  total: number;
  page: number;
  size: number;
  pages: number;
}